### DISCUSSION TITLE
Semester 4 Woche 4

### DISCUSSION THREAD

**Discussion by TimSchnur** [Christian-2003/CrInGE#44] [Student Exchange]
Hallo zusammen,

diese Woche haben wir eine grobe Planung für die ⚙️ Tests erstellt. Den momentanen Stand findet ihr wie immer 🔽 unten.
Diese werden im weiteren Verlauf der 📅 Kalenderwoche noch in den Testbericht integriert und dort ausführlich formuliert.
Auch wurde die dafür benötigte 🌐 Testumgebung bereits im Git und Lokal via Maven eingerichtet.

Des weiteren haben wir, wie bereits angekündigt, nur 📉 geringe Fortschritte in der Weiterentwicklung gemacht.

Das wäre dann auch schon alles für diese Woche. Im nächsten Blog werden wir euch weiter über das Projekt informieren.

## Testplan
Im folgendem werden die Pläne in bezug auf Tests festgelegt.

### Umfang der Tests
Wir werden ausschließlich Unit Tests verwenden.
Dabei soll auf korrekte Funktion geprüft werden.
Auf Methoden, wie Integrationstest, Validierungstests und Systemtests wird verzichtet.

### Testabdeckung
Die Testabdeckung soll einen Wert von circa 50% erreichen.

### Testwerkzeug
Zur Durchführung der Unittests wird JUnit verwendet.
Zur Ermittlung der Testabdeckung steht noch kein Werkzeug fest.

### Verwaltung der Testfälle
Die Tests werden ebenso wie der Quellcode im Git abgelegt.
Jeder Push in master soll alle Testfälle erfüllen.

**Comment by MelonigeMelone** [comment]
Hallo zusammen,
wie ich sehe habt ihr euch bereits einen Plan erstellt und die groben Rahmenbedingungen festgelegt und auch schon eure Testumgebung eingerichtet. Ich bin gespannt auf die ausführliche Version eueres Testberichts und auch auf die Integration eures ersten Tests. 
Wünsche euch weiterhin Viel Erfolg.

**Comment by lukpr0** [comment]
Hallo Cringe Team, ihr schreibt, jeder push auf den Master soll alle Testfälle erfüllen, ich nehme also an, das läuft automatisch? Außerdem würde mich noch interessieren, welche Art von Test ihr Verwendet? (Blackbox-Test, Whitebox-Test)
